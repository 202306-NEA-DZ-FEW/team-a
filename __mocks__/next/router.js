const useRouter = jest.spyOn(require("next/router"), "useRouter");

useRouter.mockImplementation(() => ({
  route: "/",
  push: "/",
  // Add other properties or methods you need for your tests.
}));

export { useRouter };
