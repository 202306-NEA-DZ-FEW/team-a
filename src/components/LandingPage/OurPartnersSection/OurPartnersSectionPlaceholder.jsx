import Container from "@/components/container";

function OurPartnersSectionPlaceholder() {
  const arrayOne = [1, 2, 3, 4, 5, 6];

  return (
    <Container className='text-center mt-24 flex flex-col items-center justify-center gap-10'>
      <div className='w-[300px] h-[35px] rounded-full bg-gray-300 animate-pulse mb-8'></div>
      <div className='flex gap-4 max-w-4xl overflow-hidden'>
        {arrayOne.map((i) => (
          <div
            key={i}
            className='w-[180px] h-[75px] rounded-md bg-gray-300 animate-pulse'
          ></div>
        ))}
      </div>
      <div className='flex gap-4 max-w-4xl overflow-hidden'>
        {arrayOne.map((i) => (
          <div
            key={i}
            className='w-[180px] h-[75px] rounded-md bg-gray-300 animate-pulse'
          ></div>
        ))}
      </div>
    </Container>
  );
}

export default OurPartnersSectionPlaceholder;
