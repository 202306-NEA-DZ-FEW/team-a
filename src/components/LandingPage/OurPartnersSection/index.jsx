import Container from "@/components/container";
import Image from "next/image";
import recoded from "public/images/recoded3.png";
import mepi from "public/images/mepi1.png";
import microsoft from "public/images/microsoft4.png";
import flatironschool from "public/images/flatironschool2.png";

function OurPartnersSection() {
  return (
    <Container className='min-h-screen flex flex-col justify-center items-center gap-8  '>
      <h1 className='text-3xl font-poppins font-bold '>Our Partners </h1>
      <div className='flex justify-center items-center flex-wrap gap-6 ms:flex-col'>
        <Image
          src={recoded}
          alt='recoded'
          width={200}
          height={100}
          className=' object-cover rounded-xl '
        />
        <Image
          src={flatironschool}
          alt='flatironschool'
          width={200}
          height={100}
          className=' object-cover rounded-xl'
        />
        <Image
          src={microsoft}
          alt='microsoft'
          width={200}
          height={100}
          className=' object-cover rounded-xl'
        />
        <Image
          src={mepi}
          alt='mepi'
          width={200}
          height={100}
          className=' object-cover rounded-xl'
        />
      </div>
    </Container>
  );
}

export default OurPartnersSection;
