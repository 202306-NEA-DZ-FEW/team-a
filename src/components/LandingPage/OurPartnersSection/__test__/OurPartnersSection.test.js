import renderer from "react-test-renderer";
import "../../../../../__mocks__/next/intersectionObserverMock";

import OurPartnersSection from "..";

jest.mock("react-slick", () => ({ children }) => <div>{children}</div>);

it("rend correctement", () => {
  const tree = renderer.create(<OurPartnersSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
