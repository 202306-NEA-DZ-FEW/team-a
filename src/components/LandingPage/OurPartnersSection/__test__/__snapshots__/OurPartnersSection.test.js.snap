// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rend correctement 1`] = `
<div
  className="max-w-6xl mx-auto px-8 xl:px-0 lg:min-h-screen my-20 lg:my-0 flex flex-col justify-center items-center gap-8  "
>
  <h1
    className="text-3xl font-poppins font-bold "
  >
    landingPage:partners
  </h1>
  <div
    className="flex justify-center items-center flex-wrap gap-6 ms:flex-col"
  >
    <img
      alt="recoded"
      className=" object-cover rounded-xl "
      data-nimg="1"
      decoding="async"
      height={100}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={200}
    />
    <img
      alt="flatironschool"
      className=" object-cover rounded-xl"
      data-nimg="1"
      decoding="async"
      height={100}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={200}
    />
    <img
      alt="microsoft"
      className=" object-cover rounded-xl"
      data-nimg="1"
      decoding="async"
      height={100}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={200}
    />
    <img
      alt="mepi"
      className=" object-cover rounded-xl"
      data-nimg="1"
      decoding="async"
      height={100}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={200}
    />
  </div>
</div>
`;
